index.js

```diff
@@ -512,41 +512,77 @@ const createJavaBom = async (
     let SBT_CMD = "sbt";
     let tempDir = fs.mkdtempSync(pathLib.join(os.tmpdir(), "cdxsbt-"));
     let tempSbtgDir = fs.mkdtempSync(pathLib.join(os.tmpdir(), "cdxsbtg-"));
-    tempSbtgDir = pathLib.join(tempSbtgDir, "1.0", "plugins");
     fs.mkdirSync(tempSbtgDir, { recursive: true });
-    // Create temporary global plugins
-    fs.writeFileSync(
-      pathLib.join(tempSbtgDir, "build.sbt"),
-      `addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.10.0-RC1")\n`
+    // Create temporary plugins file
+    let tempSbtPlugins = pathLib.join(tempSbtgDir, "dep-plugins.sbt")
+    // TODO: change to official forked version once it is available/
+    // Requires `--append` for `toFile` subtask.
+    fs.writeFileSync(tempSbtPlugins,
+      `addSbtPlugin("com.michaelpollmeier" % "sbt-dependency-graph" % "0.10.0-RC1+8-7f17b203+20210129-2104")\n`
     );
-    for (let i in sbtFiles) {
-      const f = sbtFiles[i];
-      const basePath = pathLib.dirname(f);
-      let dlFile = pathLib.join(tempDir, "dl-" + i + ".tmp");
-      console.log(
-        "Executing",
-        SBT_CMD,
-        "dependencyList in",
-        basePath,
-        "using plugins",
-        tempSbtgDir
-      );
-      const result = spawnSync(
-        SBT_CMD,
-        ["--sbt-dir", tempSbtgDir, `dependencyList::toFile"${dlFile}"`],
-        { cwd: basePath, encoding: "utf-8" }
-      );
-      if (result.status == 1 || result.error) {
+
+    // 1. Run dependency analysis on a root sbt file
```

Comment:
Can we use the flag -r to determine this behavior instead of changing the default. -r is now more or less supported for all languages. The only exception being maven where a dedicated aggregateBom step is used for performance reasons.

Reply:
I believe it's a wrong behavior because you treat sbt files as individual files. this will quickly fall apart if they have dependencies between themselves or global settings in the parent project. I imagine for similar reasons poms would fail as well

Reply:
It is possible it is a wrong behavior but is a breaking change and I have no idea how many users would get affected. Other than the performance I assume sbt would do the right thing by either using proper globals or erroring out. We handle duplicates as well as errors so there should be no issues right?

Reply:
correct